examples/ingest/s3-small-batch/main.py

```diff
@@ -0,0 +1,53 @@
+import multiprocessing as mp
+import os
+from unstructured.ingest.connector.s3_connector import S3Connector, SimpleS3Config
+from unstructured.ingest.doc_processor.generalized import process_document
+
+class MainProcess:
+
+    def __init__(self, doc_connector, doc_processor_fn, num_processes):
+        # initialize the reader and writer
+        self.doc_connector = doc_connector
+        self.doc_processor_fn = doc_processor_fn
+        self.num_processes = num_processes
+        
+
+    def initialize(self):
+        """Slower initialization things: check connections, load things into memory, etc."""
+        self.doc_connector.initialize()
+
+    def cleanup(self):
+        self.doc_connector.cleanup()
+
+    def run(self):
+        self.initialize()
+
+        self.doc_connector.fetch_docs()
+
+        # fetch the list of lazy downloading IngestDoc obj's
+        docs = self.doc_connector.fetch_docs()
+
+        # Debugging tip: use this instead of the mp.Pool loop
+        #self.doc_processor_fn(docs[0])
```

Comment:
Reminder to remove this comment

Reply:
this was intentionally left as a hint to future users

Reply:
updated comment to make that more clear